fastlane/fastlane
# Adding support for GitLab Secure Files as a Match storage mode

### Checklist

### Motivation and Context

GitLab recently released a feature called [Project-level Secure Files](https://docs.gitlab.com/ee/ci/secure_files/) which is a way to securely store configuration type files to be used in CI/CD pipelines and would work well for provisioning profiles and signing certificates.

### Description

This change introduces support for GitLab [Project-level Secure Files](https://docs.gitlab.com/ee/ci/secure_files/) as a storage backend for fastlane match. This is enabled by adding the following configuration to the Matchfile:

```
storage_mode("gitlab_secure_files")
gitlab_project('gitlab/project/path')
```

I realize this is a fairly large PR, so please let me know what additional details I can provide to make this easier to review, or of course any questions that come up along the way. Thanks!

### Testing Steps

Testing this feature can be done by setting up a project in GitLab and adding files as described in the instructions for  [Project-level Secure Files](https://docs.gitlab.com/ee/ci/secure_files/). Once the files are uploaded you can run `fastlane match` in a CI pipeline.

`fastlane match` can also be run locally by creating a personal access token in GitLab and setting the `PRIVATE_TOKEN` environment variable locally. 

I've also have a test project using this branch which you can see here https://gitlab.com/gitlab-org/incubation-engineering/devops-for-mobile-apps/gitlab_unfiltered, and the matchfile is here https://gitlab.com/gitlab-org/incubation-engineering/devops-for-mobile-apps/gitlab_unfiltered/-/blob/master/ios/fastlane/Matchfile
